examples/lines_bars_and_markers/step_demo.py

```diff
@@ -29,6 +29,22 @@
 plt.legend(title='Parameter where:')
 plt.show()
 
+# Plotting with where='between'/'edges'
+values = np.array([6, 14, 32, 37, 48, 32, 21,  4])  # hist
+edges = np.array([1., 2., 3., 4., 5., 6., 7., 8., 9.])  # bins
+fig, axes = plt.subplots(3, 2)
+axes = axes.flatten()
+axes[0].step(edges, values, where='between')
+axes[1].step(values, edges, where='between')
+axes[2].step(edges, values, where='edges')
+axes[3].step(values, edges, where='edges')
+axes[4].step(edges, values, where='edges')
+axes[4].semilogy()
+axes[5].step(edges, values, where='edges')
+axes[5].semilogy()
+
+fig.show()
```

Comment:
Would it not make sense to integrate the two new options into the same existing plot, to be able to compare them? Then for anything that does not fit here (possibly the vertical/horizontal comparisson?), start a new section only showing that bit?

<hr>

We would use `plt.show()` at the end (`fig.show()` does not necessarily do anything, if not in interactive mode).

Reply:
Isn't there value in keeping it synced with tests?

Reply:
Maybe it looks better like this.